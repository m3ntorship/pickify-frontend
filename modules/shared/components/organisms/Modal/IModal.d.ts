declare namespace IModal {
  export interface IProps {
    closeModalHandler: () => void;
  }
}

export { IModal };
