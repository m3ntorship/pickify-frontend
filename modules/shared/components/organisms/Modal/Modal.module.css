.layer {
  @apply flex justify-center items-start absolute inset-0 w-full h-full bg-dark bg-opacity-50 z-20 px-4;
}

.modal {
  @apply z-10 transform translate-y-48 w-39xl;
}
