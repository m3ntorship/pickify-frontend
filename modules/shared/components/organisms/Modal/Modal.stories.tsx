import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import Modal from './Modal';
import type { IModal } from './IModal';

export default {
  title: 'organisms/Modal',
  component: Modal,
};

const Template: Story<IModal.IProps> = (args): ReactElement => (
  <Modal {...args} />
);
export const Default = Template.bind({});
Default.args = {};
