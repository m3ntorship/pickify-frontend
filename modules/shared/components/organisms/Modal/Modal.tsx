import type { FC, ReactElement, MouseEvent } from 'react';
import type { IModal } from './IModal';

const Modal: FC<IModal.IProps> = ({
  children,
  closeModalHandler,
}): ReactElement => {
  const preventClickHandler = (e: MouseEvent<HTMLDivElement>): void => {
    e.stopPropagation();
  };

  return (
    <div
      className="flex justify-center items-start absolute inset-0 w-full h-full bg-dark bg-opacity-50 z-20 px-4"
      onClick={closeModalHandler}
      aria-hidden
      role="banner"
    >
      <div
        className="z-10 transform translate-y-48"
        onClick={preventClickHandler}
        aria-hidden
        role="treeitem"
      >
        {children}
      </div>
    </div>
  );
};
export default Modal;
