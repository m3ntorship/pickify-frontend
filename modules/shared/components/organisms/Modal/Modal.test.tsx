import React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Modal from './Modal';

describe('Modal', () => {
  it('should render modal component with children elements', () => {
    const mockedFn = jest.fn();
    const tree = renderer
      .create(
        <Modal closeModalHandler={mockedFn}>
          <div>Modal</div>
        </Modal>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call closeModalHandler function', () => {
    const mockedFn = jest.fn();
    const calledOnce = 1;
    render(
      <Modal closeModalHandler={mockedFn}>
        <div>Modal</div>
      </Modal>,
    );

    const layer = screen.getByTestId('layer');

    userEvent.click(layer);

    expect(mockedFn).toHaveBeenCalledTimes(calledOnce);
  });
});
