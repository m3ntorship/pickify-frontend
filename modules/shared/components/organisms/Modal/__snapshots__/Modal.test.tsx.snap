// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal component with children elements 1`] = `
<div
  aria-hidden={true}
  className="layer"
  data-testid="layer"
  onClick={[MockFunction]}
  role="banner"
>
  <div
    aria-hidden={true}
    className="modal"
    onClick={[Function]}
    role="treeitem"
  >
    <div>
      Modal
    </div>
  </div>
</div>
`;
